zingocli/src/lib.rs

```diff
@@ -30,31 +49,36 @@ pub fn build_clap_app() -> clap::ArgMatches {
                 .action(clap::ArgAction::SetTrue))
             .arg(Arg::new("chain")
                 .long("chain").short('c')
-                .help(r#"What chain to expect, if it's not inferable from the server URI. One of "mainnet", "testnet", or "regtest""#))
-            .arg(Arg::new("from")
-                .short('f')
-                .short_alias('s')
-                .long("from")
-                .alias("seed")
-                .alias("viewing-key")
-                .value_name("from")
-                .value_parser(parse_seed)
-                .help("Create a new wallet with the given key. Can be a 24-word seed phrase or a viewkey. Will fail if wallet already exists"))
+                .help(r#"What chain to expect, if it's not inferrable from the server URI. One of "mainnet", "testnet", or "regtest""#))
+            .arg(Arg::new("seed-phrase")
+                .long("seed-phrase")
+                .hide(true)
+                .env("SEED_PHRASE")
+                .value_parser(WalletBase::parse_input_to_phrase)
+                .requires("birthday")
+                .help("Create a new wallet with the given seed phrase. Cannot be used with --view-key, or --wallet-dir."))
+            .arg(Arg::new("view-key")
+                .long("view-key")
+                .hide(true)
+                .env("VIEW_KEY")
+                .value_parser(ufvk_to_string_helper)
+                .requires("birthday")
+                .help("Create a new wallet with the given viewkey. Cannot be used with --seed-phrase, or --wallet-dir."))
             .arg(Arg::new("birthday")
                 .long("birthday")
                 .value_name("birthday")
                 .value_parser(clap::value_parser!(u32))
-                .help("Specify wallet birthday when restoring from seed. This is the earliest block height where the wallet has a transaction."))
+                .help("Specify wallet birthday when restoring from seed. This is the earlist block height where the wallet has a transaction."))
+            .arg(Arg::new("wallet-dir")
+                .long("wallet-dir")
+                .value_name("wallet-dir")
+                .help("Launch with an existing wallet specified by the value which is the absolute path to the wallet. Cannot be used with --seed-phrase, or --view-key."))
```

Comment:
Why can't this be used with `--seed-phrase` or `--view-key`? How do you specify what directory to create the new wallet in?

Reply:
It's because the `wallet-dir` arg is overloaded.  If one exists it is used as the source of truth to load from..  which is mutually exclusive with passing a seed-phrase or view-key.

Really there should be an input-dir which is mutually exclusive with other inputs.

Reply:
I think the current architecture of "if wallet-dir has wallet, load wallet (incompatible with seed-phrase/view-key), else create wallet in that location" is functional, and (if I understand the new architecture correctly) "if creating a wallet from seed/viewkey, it's not possible to specify a wallet location" isn't.

Reply:
Try something like:

`cargo run -- --view-key="uview1gtxrj97cdrlf7u4c7pqa8qhsrle5y3uxd9axfrn40e6j4j6grct00xmns4tcfetg8dsl8hq5znu8pz60syrsxxxwc7m479grsfxj2nk7c0ef2k9xra5erf3nawn7qyzp585xfjqtzs4gcg53tdqpsladzezy2h9mdm75ud4wthnd9kj8fyel42x296clgr3dm4lwvduxly686z2npqhuyjcgs8adw7rpxh9f9s5tjyaj43fzxymv0fd6w868swuzwpay29xmrv3mx7htqeevehxd7e25z8ktknvwl6nmyz3j7sk4shu9nqqntskk9y4yadd8e0e65nhmd62h38ld3sxwwkmy2cv53lfngccnmlmvuhdqujdaek6hlsrzasut53pma9nenqcz6xlqag0w0wvqulyfs4as2vfmxl98k6gwr8ch659rcm0lhveyfk7mg0j3ldkqf4v55yat7938a7gj5ha530r57hpkruphj5sxnmsuqczsa0sq" --birthday=2363649 --fresh-wallet=./FOOOOOO`

Reply:
OR

`cargo run -- --help`